js/spreadsheet.js

```diff
@@ -861,16 +840,32 @@ var spreadsheet = function(options) {
     return entries;
   };
 
-  function validateJsonString(str) {
-    var validatedString;
-
+  function getColumnValue(str) {
     try {
-      validatedString = jsonObjRegExp.test(str) ? JSON.parse(str) : str;
+      var parsingResult = JSON.parse('[' + str + ']');
```

Comment:
@OleksandrLavrynenko Based on what you're saying, an array of objects are joined using `,` and printed on Handsontable, and that's why [{foo: 1}] --> '{"foo": 1}'. Is that right?

If so, then it sounds like it shouldn't be displayed by joining with `,`. If we have the right capability to parse and display JSON, then it can just be stringified before displaying with Handsontable, and parsed into JSON before it's saved.

Would that solve the problem?

Reply:
**@tonytlwu** I converted arrays and objects to JSON for display. Are you satisfied with this format for display arrays and objects?
![obj-arr](https://user-images.githubusercontent.com/86256215/140922994-1439e4a2-3066-4567-a64f-c38ed479782b.jpg)

Reply:
@OleksandrLavrynenko This looks good. How does it deal with other types, e.g. string, integers, booleans.

Reply:
**@tonytlwu** strings, numbers and booleans are converted to a string after editing or at the moment of entering them into the table, just as it works now.

Reply:
@OleksandrLavrynenko Can you show screenshots with the examples of the various types just as you did in your previous comments please?

Reply:
**@tonytlwu** Here is a screenshot of the sent data and its output in the table
![types](https://user-images.githubusercontent.com/86256215/140990768-9660b48a-a63b-4981-aead-f09a17740ee8.jpg)
![arr](https://user-images.githubusercontent.com/86256215/140990770-9914d689-9717-4e39-8bf7-7eefd8ae2137.jpg)

Reply:
@OleksandrLavrynenko Looks good to me. Please continue with the solution.